builder/kubernetes/platform.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kubernetes
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/tsuru/tsuru/builder"
+	tsuruErrors "github.com/tsuru/tsuru/errors"
+	"github.com/tsuru/tsuru/provision"
+	appTypes "github.com/tsuru/tsuru/types/app"
+)
+
+var _ builder.Builder = &kubernetesBuilder{}
+
+func (b *kubernetesBuilder) PlatformAdd(opts appTypes.PlatformOptions) error {
+	return b.buildPlatform(opts.Name, opts.Args, opts.Output, opts.Data, opts.Ctx)
+}
+
+func (b *kubernetesBuilder) PlatformUpdate(opts appTypes.PlatformOptions) error {
+	return b.buildPlatform(opts.Name, opts.Args, opts.Output, opts.Data, opts.Ctx)
+}
+
+func (b *kubernetesBuilder) PlatformRemove(name string) error {
+	return nil
+}
+
+func (b *kubernetesBuilder) buildPlatform(name string, args map[string]string, w io.Writer, data []byte, ctx context.Context) error {
+	client, err := getKubeClient()
+	if err != nil {
+		return err
+	}
+	var buf bytes.Buffer
+	writer := tar.NewWriter(&buf)
```

Comment:
I think we could extract the tar creation logic to a common place shared by the builders